quantum/os_detection.c

```diff
@@ -45,41 +48,100 @@ struct setups_data_t setups_data = {
     .cnt_ff = 0,
 };
 
-os_variant_t detected_os = OS_UNSURE;
+#    ifndef OS_DETECTION_DEBOUNCE
+#        define OS_DETECTION_DEBOUNCE 200
+#    endif
 
-// Some collected sequences of wLength can be found in tests.
-void make_guess(void) {
-    if (setups_data.count < 3) {
-        return;
-    }
-    if (setups_data.cnt_ff >= 2 && setups_data.cnt_04 >= 1) {
-        detected_os = OS_WINDOWS;
-        return;
-    }
-    if (setups_data.count == setups_data.cnt_ff) {
-        // Linux has 3 packets with 0xFF.
-        detected_os = OS_LINUX;
-        return;
-    }
-    if (setups_data.count == 5 && setups_data.last_wlength == 0xFF && setups_data.cnt_ff == 1 && setups_data.cnt_02 == 2) {
-        detected_os = OS_MACOS;
-        return;
-    }
-    if (setups_data.count == 4 && setups_data.cnt_ff == 0 && setups_data.cnt_02 == 2) {
-        // iOS and iPadOS don't have the last 0xFF packet.
-        detected_os = OS_IOS;
-        return;
+// 2s should always be more than enough
+#    if OS_DETECTION_DEBOUNCE > 2000
+#        undef OS_DETECTION_DEBOUNCE
+#        define OS_DETECTION_DEBOUNCE 2000
+#    endif
+
+#    ifndef OS_DETECTION_KEYBOARD_RESET
+#        define OS_DETECTION_KEYBOARD_RESET true
+#    endif
+
+#    ifndef OS_DETECTION_KEYBOARD_RESET_BOOTLOADER
+#        define OS_DETECTION_KEYBOARD_RESET_BOOTLOADER false
+#    endif
```

Comment:
I'm not a fan of setting this to true/false. defined or not should be fine
```suggestion
```

Reply:
Done.